### PR TITLE
added secrets to distributed swarm minio

### PR DESCRIPTION
**Pull request by morph027**
## Description

I've just added the supports for swarm secrets to the docker compose file for distributed minio.

## Motivation and Context

Secrets are just better than env. vars.

## How Has This Been Tested?

* created secrets according to https://docs.minio.io/docs/minio-docker-quickstart-guide
* deployed stack

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Contrib/3rdparty improvement

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by nitisht** [comment]
@morph027 This looks great. Could you also add details on how to run this in the Swarm guide - https://github.com/minio/minio/blob/master/docs/orchestration/docker-swarm/README.md

**Comment by morph027** [comment]
Jip, coming up.

**Review by krisis — APPROVED** [review]
LGTM. Thank you @morph027 for the PR.

**Review by harshavardhana — DISMISSED** [review]

**Comment by harshavardhana** [comment]
@NitishT any other changes needed ?

**Comment by nitisht** [comment]
@harshavardhana I need to try it out locally once, will update once done.

**Comment by deekoder** [comment]
@NitishT i will merge this after you complete testing.

File: docs/orchestration/docker-swarm/README.md [hunk 1]
```diff
@@ -24,13 +24,21 @@ docker swarm init --advertise-addr <MANAGER-IP>
...
+## 3. Create Docker secrets for Minio
+
+```shell
+echo "AKIAIOSFODNN7EXAMPLE" | docker secret create access_key -
+echo "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY" | docker secret create secret_key -
+```
+
+## 4. Deploy distributed Minio services
 
 Download the [Docker Compose file](https://github.com/minio/minio/blob/master/docs/orchestration/docker-swarm/docker-compose.yaml?raw=true) on your Swarm master. Then execute the command
```
Comment (nitisht) [inline docs/orchestration/docker-swarm/README.md]: Can you change the `Docker Compose file` link to the new compose file as well.

File: docs/orchestration/docker-swarm/README.md [hunk 2]
```diff
@@ -24,13 +24,21 @@ docker swarm init --advertise-addr <MANAGER-IP>
...
 
 ```shell
 docker stack deploy --compose-file=docker-compose.yaml minio_stack
```
Comment (nitisht) [inline docs/orchestration/docker-swarm/README.md]: Command line argument needs to be changed as well.

**Review by nitisht — CHANGES_REQUESTED** [review]
@morph027 When I tried running the command `docker stack deploy --compose-file=docker-compose-secrets.yaml minio_stack` (with the new file) 

I get the error `secrets Additional property secrets is not allowed`. Can you help me with what is wrong here.

**Review by harshavardhana — APPROVED** [review]

**Comment by nitisht** [comment]
Thanks for the update @morph027 works fine now. LGTM

**Review by nitisht — APPROVED** [review]

**Pull request merged**